syncall/google/gtasks_side.py

```diff
@@ -0,0 +1,294 @@
+            # object to the list_next method.
+            request = self._service.tasks().list_next(request, response)  # type: ignore
+
+        # cache them
+        for t in tasks:
+            self._items_cache[t["id"]] = t
+
+        return tasks
+
+    def get_item(self, item_id: str, use_cached: bool = True) -> Optional[dict]:
+        item = self._items_cache.get(item_id)
+        if not use_cached or item is None:
+            item = self._get_item_refresh(item_id=item_id)
+
+        return item
+
+    def _get_item_refresh(self, item_id: str) -> Optional[dict]:
+        ret = None
+        try:
+            ret = (
+                self._service.tasks().get(tasklist=self._task_list_id, task=item_id).execute()
+            )
+            if ret["status"] == "deleted":
+                ret = None
+                self._items_cache.pop(item_id)
+            else:
+                self._items_cache[item_id] = ret
+        except HttpError:
+            pass
+        finally:
+            return ret
+
+    def update_item(self, item_id, **changes):
+        # Check if item is there
+        task = self._service.tasks().get(tasklist=self._task_list_id, task=item_id).execute()  # type: ignore
+        task.update(changes)
+        self._service.tasks().update(  # type: ignore
+            tasklist=self._task_list_id, task=task["id"], body=task
+        ).execute()
+
+    def add_item(self, item) -> dict:
+        task = self._service.tasks().insert(tasklist=self._task_list_id, body=item).execute()  # type: ignore
+        logger.debug(f'Task created -> {task.get("selfLink")}')
+
+        return task
+
+    def delete_single_item(self, item_id) -> None:
+        self._service.tasks().delete(tasklist=self._task_list_id, task=item_id).execute()  # type: ignore
+
+    @classmethod
+    def id_key(cls) -> str:
+        return cls.ID_KEY
+
+    @classmethod
+    def summary_key(cls) -> str:
+        return cls.TITLE_KEY
+
+    @classmethod
+    def last_modification_key(cls) -> str:
+        return cls.LAST_MODIFICATION_KEY
+
+    @staticmethod
+    def get_task_due_time(item: dict) -> datetime.datetime:
+        """
+        Return the datetime on which task is due in datetime format.
+
+        """
+        dt = GTasksSide.parse_datetime(item["due"])
+        return dt
+
+    @staticmethod
+    def get_task_completed_time(item: dict) -> Union[datetime.datetime, None]:
+        """
+        Return the datetime on which task was completed in datetime format.
+
+        """
+        if "completed" in item.keys():
+            dt = GTasksSide.parse_datetime(item["completed"])
+
+            assert isinstance(dt, datetime.datetime)
+            return dt
+
+    @staticmethod
+    def format_datetime(dt: datetime.datetime) -> str:
+        assert isinstance(dt, datetime.datetime)
+        return format_datetime_tz(dt)
+
+    @classmethod
+    def parse_datetime(cls, dt: Union[str, dict, datetime.datetime]) -> datetime.datetime:
```

Comment:
This looks like a duplicate of the function we have in GCalSide. Could we abstract this? Perhaps put it in the parent class?